packages/notification-user-lifecycle-consumer/src/services/messageProcessor.ts

```diff
@@ -0,0 +1,182 @@
+import { InteropTokenGenerator, logger } from "pagopa-interop-commons";
+import {
+  unsafeBrandId,
+  genericInternalError,
+  generateId,
+} from "pagopa-interop-models";
+import type { UserId, SelfcareId, CorrelationId } from "pagopa-interop-models";
+import { EachMessagePayload } from "kafkajs";
+import { match } from "ts-pattern";
+import { notificationConfigApi } from "pagopa-interop-api-clients";
+import { UsersEventPayload } from "../model/UsersEventPayload.js";
+import { config } from "../config/config.js";
+import { ReadModelServiceSQL } from "./readModelServiceSQL.js";
+import { UserServiceSQL } from "./userServiceSQL.js";
+
+// eslint-disable-next-line max-params
+export async function processUserEvent(
+  payload: UsersEventPayload,
+  readModelServiceSQL: ReadModelServiceSQL,
+  userServiceSQL: UserServiceSQL,
+  notificationConfigProcessClient: ReturnType<
+    typeof notificationConfigApi.createProcessApiClient
+  >,
+  interopTokenGenerator: InteropTokenGenerator,
+  loggerInstance: ReturnType<typeof logger>,
+  correlationId: CorrelationId
+): Promise<void> {
+  const userId = unsafeBrandId<UserId>(payload.user.userId);
+  const institutionId = unsafeBrandId<SelfcareId>(payload.institutionId);
+  const action = payload.eventType;
+  const tenantId = await readModelServiceSQL.getTenantIdBySelfcareId(
+    institutionId
+  );
+
+  if (!tenantId) {
+    throw genericInternalError(
+      `Tenant not found for selfcareId: ${institutionId}`
+    );
+  }
+
+  const userData = {
+    userId,
+    tenantId,
+    institutionId,
```

Comment:
I'm not sure if it would be clearer to use the internal name `selfcareId` (or rather `tenantSelfcareId`) since in the tenant model we call this `selfcareId` and not `institutionId`